crates/oxc_regular_expression/src/display.rs

```diff
@@ -0,0 +1,368 @@
+
+impl<'a> Display for RegularExpression<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "/{}/{}", self.pattern, self.flags)
+    }
+}
+
+impl Display for Flags {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut flags = String::with_capacity(8);
+        macro_rules! if_true_append {
+            ($flag:ident, $char:literal) => {
+                if self.$flag {
+                    flags.push($char);
+                }
+            };
+        }
+
+        // write flags in the order they are described in the `MDN`
+        // <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#advanced_searching_with_flags>
+        if_true_append!(has_indices, 'd');
+        if_true_append!(global, 'g');
+        if_true_append!(ignore_case, 'i');
+        if_true_append!(multiline, 'm');
+        if_true_append!(dot_all, 's');
+        if_true_append!(unicode, 'u');
+        if_true_append!(unicode_sets, 'v');
+        if_true_append!(sticky, 'y');
+
+        write!(f, "{flags}")
+    }
+}
+
+impl<'a> Display for Pattern<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.body)
+    }
+}
+
+impl<'a> Display for Disjunction<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_join(f, "|", &self.body)
+    }
+}
+
+impl<'a> Display for Alternative<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_join(f, "", &self.body)
+    }
+}
+
+impl<'a> Display for Term<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::BoundaryAssertion(it) => write!(f, "{it}"),
+            Self::LookAroundAssertion(it) => write!(f, "{}", it.as_ref()),
+            Self::Quantifier(it) => write!(f, "{}", it.as_ref()),
+            Self::Character(it) => write!(f, "{it}"),
+            Self::Dot(it) => write!(f, "{it}"),
+            Self::CharacterClassEscape(it) => write!(f, "{it}"),
+            Self::UnicodePropertyEscape(it) => write!(f, "{}", it.as_ref()),
+            Self::CharacterClass(it) => write!(f, "{}", it.as_ref()),
+            Self::CapturingGroup(it) => write!(f, "{}", it.as_ref()),
+            Self::IgnoreGroup(it) => write!(f, "{}", it.as_ref()),
+            Self::IndexedReference(it) => write!(f, "{it}"),
+            Self::NamedReference(it) => write!(f, "{}", it.as_ref()),
+        }
+    }
+}
+
+impl Display for BoundaryAssertion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.kind)
+    }
+}
+
+impl Display for BoundaryAssertionKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Start => write!(f, "^"),
+            Self::End => write!(f, "$"),
+            Self::Boundary => write!(f, "\\b"),
+            Self::NegativeBoundary => write!(f, "\\B"),
```

Comment:
```suggestion
            Self::Boundary => write!(f, r"\b"),
            Self::NegativeBoundary => write!(f, r"\B"),
```

This is easier on the eyes...? 👀